modules/authentication/snip-enabling-user-authentication-with-github-optional-authentication-provider-steps.adoc

```diff
@@ -0,0 +1,59 @@
+:_mod-docs-content-type: SNIPPET
+Optional: Consider adding optional fields to the `auth.providers.github` section in your `{my-app-config-file}` file:
+
+[source,yaml,subs="+quotes"]
+----
+auth:
+  environment: production
+  providers:
+    github:
+      production:
+        clientId: ${GITHUB_CLIENT_ID}
+        clientSecret: ${GITHUB_CLIENT_SECRET}
+        callbackUrl: __<your_intermediate_service_url/handler>__
+        sessionDuration: { hours: 24 }
+        signIn:
+          resolvers:
+            - resolver: usernameMatchingUserEntityName
+              dangerouslyAllowSignInWithoutUserInCatalog: true
+signInPage: github
+----
+
+`callbackUrl`::
+Enter the callback URL that GitHub uses when initiating an OAuth flow, such as: __<your_intermediate_service_url/handler>__.
+Define it when {product-short} is not the immediate receiver, such as in cases when you use one OAuth app for many {product-short} instances.
+
+`sessionDuration`::
+Enter the user session lifespan, in `ms` library format (such as '24h', '2 days'), ISO duration, or "human duration".
+
+`signIn`::
+
+`resolvers`:::
+After successful authentication, {product-short} resolves the user signing in to an existing user in the {product-short} catalog.
+To best match users securely for your use case, consider configuring a specific resolver.
++
+Enter the resolver list to override the default resolver: `usernameMatchingUserEntityName`.
++
+The authentication provider tries each sign-in resolver in order until it succeeds, and fails if none succeed.
```

Comment:
```suggestion
The authentication provider tries each sign-in resolver in order until it succeeds. If none of the attempts succeed, the sign-in fails.
```

Reply:
this might be duplication because technically speaking, it's obvious that it fails if it doesn't succeed